crates/node/src/networking/p2p/protocol/v1.rs

```diff
@@ -0,0 +1,183 @@
+use eyre::Result;
+use std::{collections::BTreeSet, net::SocketAddr};
+
+use gevulot_node::types;
+use libsecp256k1::PublicKey;
+use serde::{Deserialize, Serialize};
+
+use super::internal;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum Handshake {
+    V1(HandshakeV1),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct HandshakeV1 {
+    pub my_p2p_listen_addr: SocketAddr,
+    pub peers: BTreeSet<SocketAddr>,
+    pub http_port: Option<u16>,
+}
+
+impl From<internal::Handshake> for Handshake {
+    fn from(value: internal::Handshake) -> Self {
+        let internal::Handshake {
+            my_p2p_listen_addr,
+            peers,
+            http_port,
+        } = value;
+        Handshake::V1(HandshakeV1 {
+            my_p2p_listen_addr,
+            peers,
+            http_port,
+        })
+    }
+}
+
+impl Handshake {
+    pub fn parse(bs: &[u8]) -> Result<internal::Handshake> {
+        match bincode::deserialize(bs) {
+            Ok(Handshake::V1(HandshakeV1 {
+                my_p2p_listen_addr,
+                peers,
+                http_port,
+            })) => Ok(internal::Handshake {
+                my_p2p_listen_addr,
+                peers,
+                http_port,
+            }),
+            Err(err) => Err(err.into()),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum Message {
+    V1(MessageV1),
+}
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum MessageV1 {
+    Transaction(types::Transaction<types::transaction::Validated>),
+    DiagnosticsRequest(DiagnosticsRequestKind),
+    DiagnosticsResponse(PublicKey, DiagnosticsResponseV1),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum DiagnosticsRequestKind {
+    Version,
+    Resources,
+    Metrics,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum DiagnosticsResponseV1 {
+    Version {
+        major: u16,
+        minor: u16,
+        patch: u16,
+        build: String,
+    },
+    Resources {
+        cpus: u64,
+        mem: u64,
+        gpus: u64,
+    },
+    Metrics(Vec<u8>),
```

Comment:
This will hold encoded prometheus metrics from the node. With this, we can later on just add new metrics and that won't require changes to P2P wire-protocol.